tests/unit/CIGitLogicTest.ts

```diff
@@ -0,0 +1,473 @@
+/**
+ * @jest-environment node
```

Comment:
Ported this from a shell script to Node.js because it made it easier to debug, and generally more flexible. Overriding `console.log` to get the output of a script (as we were doing before) made it so that we couldn't really see logs from the script without breaking the assertions we had.

It was a fair amount of work (obviously, the test is large), but it makes it considerably more flexible to work with. You can easily set breakpoints at any step of the script, checkout the directory manually, run whatever commands you want, and continue.